build.gradle

```diff
@@ -172,33 +171,27 @@ tasks.register("configureArtifactInfo") {
     description "Set the url to download stack artifacts for select stack version"
 
     doLast {
-        def versionQualifier = System.getenv('VERSION_QUALIFIER')
-        if (versionQualifier) {
-            version = "$version-$versionQualifier"
-        }
-
-        boolean isReleaseBuild = System.getenv('RELEASE') == "1" || versionQualifier
-        String apiResponse = artifactVersionsApi.toURL().text
-
-        def dlVersions = new JsonSlurper().parseText(apiResponse)
-        String qualifiedVersion = dlVersions['versions'].grep(isReleaseBuild ? ~/^${version}$/ : ~/^${version}-SNAPSHOT/)[0]
-        if (qualifiedVersion == null) {
-            if (!isReleaseBuild) {
-                project.ext.set("useProjectSpecificArtifactSnapshotUrl", true)
-                project.ext.set("stackArtifactSuffix", "${version}-SNAPSHOT")
-                return
+        def splitVersion = version.split('\\.')
+        def major = splitVersion[0].toInteger()
+        def minor = splitVersion[1].toInteger()
+        def branch = "${major}.${minor}"
+        def fallbackMajorX = "${major}.x"
+        def fallbackMinor = (minor - 1 < 0)? "${major-1}.x" : "${major}.${minor-1}"
+        def qualifiedVersion = ""
```

Comment:
```suggestion
        int major = splitVersion[0].toInteger()
        int minor = splitVersion[1].toInteger()
        String branch = "${major}.${minor}"
        String fallbackMajorX = "${major}.x"
        boolean isMajorFirstMinor = minor - 1 < 0
        String fallbackMinor = isMajorFirstMinor ? "${major-1}.x" : "${major}.${minor-1}"
        def qualifiedVersion = ""
```

Reply:
Sorry I'm a fan of types, being explicit could help in readability, plus extracted the ternary condition to explicitly states what it means.